src/API/ProductsLowInStock.php

```diff
@@ -0,0 +1,309 @@
+				$query_result->images     = $this->get_images( $product );
+				$query_result->attributes = $this->get_attributes( $product );
+				return $query_result;
+			},
+			$query_results['results']
+		);
+
+		// set last_order_date.
+		$query_results['results'] = $this->set_last_order_date( $query_results['results'] );
+
+		// convert the post data to the expected API response for the backward compatibility.
+		$query_results['results'] = array_map( array( $this, 'transform_post_to_api_response' ), $query_results['results'] );
+
+		$response = rest_ensure_response( array_values( $query_results['results'] ) );
+		$response->header( 'X-WP-Total', $query_results['total'] );
+		$response->header( 'X-WP-TotalPages', $query_results['pages'] );
+		$response->header( 'Cache-Control', 'max-age=300' );
+
+		return $response;
+	}
+
+	/**
+	 * Set the last order date for each data.
+	 *
+	 * @param array $results query result from get_low_in_stock_products.
+	 *
+	 * @return mixed
+	 */
+	protected function set_last_order_date( $results = array() ) {
+		global $wpdb;
+		if ( 0 === count( $results ) ) {
+			return $results;
+		}
+
+		$wheres = array();
+		foreach ( $results as $result ) {
+			'variation_product' === $result->post_type ?
+				array_push( $wheres, "(product_id={$result->post_parent} and variation_id={$result->ID})" )
+				: array_push( $wheres, "product_id={$result->ID}" );
+		}
+
+		count( $wheres ) ? $where_clause = implode( ' or ', $wheres ) : $where_clause = $wheres[0];
+
+		$product_lookup_table = $wpdb->prefix . 'wc_order_product_lookup';
+		$query_string         = "
+			select product_id, MAX( wc_order_product_lookup.date_created ) AS last_order_date 
+			from {$product_lookup_table} wc_order_product_lookup
+			where {$where_clause}
+			group by product_id
+			order by date_created desc
+		";
+
+		// phpcs:ignore -- ignore prepare() warning as we're not using any user input here.
+		$query_results = $wpdb->get_results( $query_string, OBJECT_K );
+		foreach ( $query_results as $key => $query_result ) {
+			if ( isset( $results[ $key ] ) ) {
+				$results[ $key ]->last_order_date = $query_result->last_order_date;
+			}
+		}
+
+		return $results;
+	}
+
+	/**
+	 * Get low in stock products data.
+	 *
+	 * @param int    $page current page.
+	 * @param int    $per_page items per page.
+	 * @param string $status post status.
+	 *
+	 * @return array
+	 */
+	protected function get_low_in_stock_products( $page = 1, $per_page = 1, $status = 'publish' ) {
+		global $wpdb;
+
+		$offset                       = ( $page - 1 ) * $per_page;
+		$low_stock_threshold          = absint( max( get_option( 'woocommerce_notify_low_stock_amount' ), 1 ) );
+		$use_sitewide_stock_threshold = $this->is_using_sitewide_stock_threshold_only();
+
+		if ( $use_sitewide_stock_threshold ) {
+			$query_string = $this->get_query( $use_sitewide_stock_threshold );
+		} else {
+			$query_string = $this->get_query();
+		}
```

Comment:
Ah sorry, before I meant refactoring the whole `if-else` block like:
```suggestion
		$query_string = $this->get_query( $use_sitewide_stock_threshold );
```

I'm happy if you want to keep the `if-else` here. In that case it’s perhaps just better to pass `true` / `false` to `get_query()` like you originally had 😅

Reply:
I can't keep `if-else` after seeing this comment. haha :) I'll make the change.

Reply:
updated in https://github.com/woocommerce/woocommerce-admin/pull/7377/commits/ca14814b2c87417b44dae95e516d3600b66dfa28